spec/services/dialog_local_service_spec.rb

```diff
@@ -41,6 +41,78 @@
     let(:button_name) { "custom-button-name" }
     let(:resource_action) { instance_double("ResourceAction", :id => 321, :dialog_id => 654) }
 
+    context "when the object is a AvailabilityZone" do
+      let(:obj) { double(:class => ManageIQ::Providers::Vmware::InfraManager::AvailabilityZone, :id => 123) }
+
+      it "returns a hash" do
+        expect(service.determine_dialog_locals_for_custom_button(obj, button_name, resource_action)).to eq(
+          :resource_action_id     => 321,
+          :target_id              => 123,
+          :target_type            => 'availability_zone',
+          :dialog_id              => 654,
+          :force_old_dialog_use   => false,
+          :api_submit_endpoint    => "/api/availability_zones/123",
+          :api_action             => "custom-button-name",
+          :finish_submit_endpoint => "/availability_zone",
+          :cancel_endpoint        => "/availability_zone"
+        )
+      end
+    end
```

Comment:
I see 15 `it` blocks doing very similar thing, just with different parameters. Could this be turned into a shared example perhaps? Or something else by using parameters?

Reply:
@mzazrivec Good call, easy fix with a shared example, will do that. 👍